.github/workflows/main.yml

```diff
@@ -1,5 +1,8 @@
 name: build-and-deploy
-on: push
+on:
+  push:
+    branches:
+      - main
```

Comment:
add `npm build` job on push event for all branches

Reply:
fixed.